ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,868 @@
+# - increase the minor number if new columns or datasets are added
+# - increase the patch number if there is a small bugfix to the model.
+DL1_DATA_MODEL_VERSION = "v1.0.0"
+
+
+def write_core_provenance(output_filename, obs_id, subarray, writer):
+    """
+    Attaches Core Provenence headers to an output HDF5 file.
+    Right now this is hard-coded for use with the ctapipe-stage1-process tool
+
+    TODO: generalize, and move to ctapipe.io
+
+    Parameters
+    ----------
+    output_filename: str
+        output HDF5 file
+    obs_id: int
+        observation ID
+    subarray:
+        SubarrayDescription to get metadata from
+    writer: HDF5TableWriter
+        output
+    """
+    import uuid
+
+    activity = PROV.current_activity.provenance
+
+    metadata = {
+        "CTA REFERENCE VERSION ": "1",
+        "CTA CONTACT ORGANIZATION": "CTA Consortium",
+        "CTA CONTACT NAME": "K. Kosack",
+        "CTA CONTACT EMAIL": "karl.kosack@cea.fr",
+        "CTA PRODUCT DESCRIPTION": "DL1 Event List",
+        "CTA PRODUCT CREATION_TIME": Time.now().iso,
+        "CTA PRODUCT ID": str(uuid.uuid4()),
+        "CTA PRODUCT DATA CATEGORY": "S",
+        "CTA PRODUCT DATA LEVEL": "DL1",
+        "CTA PRODUCT DATA TYPE": "Event",
+        "CTA PRODUCT DATA ASSOCIATION": "Subarray",
+        "CTA PRODUCT DATA MODEL NAME": "DL1/Event",
+        "CTA PRODUCT DATA MODEL VERSION": DL1_DATA_MODEL_VERSION,
+        "CTA PRODUCT DATA MODEL URL": "",
+        "CTA PRODUCT FORMAT": "hdf5",
+        "CTA PROCESS TYPE": "simulation",
+        "CTA PROCESS SUBTYPE": "",
+        "CTA PROCESS ID": str(obs_id),
+        "CTA ACTIVITY NAME": activity["activity_name"],
+        "CTA ACTIVITY TYPE": "software",
+        "CTA ACTIVITY ID": activity["activity_uuid"],
+        "CTA ACTIVITY START": activity["start"]["time_utc"],
+        "CTA ACTIVITY SOFTWARE NAME": "ctapipe",
+        "CTA ACTIVITY SOFTWARE VERSION": activity["system"]["ctapipe_version"],
+        "CTA INSTRUMENT SITE": "unknown",
+        "CTA INSTRUMENT CLASS": "subarray",
+        "CTA INSTRUMENT TYPE": "",
+        "CTA INSTRUMENT SUBTYPE": "",
+        "CTA INSTRUMENT VERSION": "",
+        "CTA INSTRUMENT ID": str(subarray),
+    }
+
+    for key, value in metadata.items():
+        writer._h5file.root._v_attrs[key] = value
+
+
+def morphology(geom, image_mask) -> MorphologyContainer:
+    """
+    Compute image morphology parameters
+    Parameters
+    ----------
+    geom: ctapipe.instrument.camera.CameraGeometry
+        camera description
+    image_mask: np.ndarray(bool)
+        image of pixels surviving cleaning (True=survives)
+    Returns
+    -------
+    MorphologyContainer:
+        parameters related to the morphology
+    """
+
+    num_islands, island_labels = number_of_islands(geom=geom, mask=image_mask)
+
+    return MorphologyContainer(num_pixels=image_mask.sum(), num_islands=num_islands)
+
+
+class IntensityContainer(Container):
```

Comment:
The use of Intensity both for the charge of individual pixels and for the total image charge seems prone to confusion: "mean intensity" or "standard deviation of intensity" can refer to pixels within an image(like here), or to images within a run.

Reply:
And you can calculate the std of the std of the pixels for each run and then the std of the std of the std over multiple runs. 

This data is stored in `/dl1/event/telescope/parameters` which makes it pretty clear, doesn't it?